Replace winston and console with pino logger.

dmitrizagidulin opened the pull request:
## Summary

In `server-core`:
- Replaces Winston logger with `pino`
- Replaces usages of `console.log` with appropriate logger statements.
- Removes usage of deprecated `feathers-logger` package 
- Also enhances `dev-agones-silent` npm script by using the cross-platform `dev-null-cli`  package

## References

Partly addresses issue #2799.

## Checklist
- [x] CI/CD checks pass `npm run check`

dmitrizagidulin commented on packages/server-core/src/media/upload-media/upload-asset.service.ts:
```diff
@@ -99,7 +99,7 @@ export default (app: Application): void => {
         if (typeof data.args === 'string') data.args = JSON.parse(data.args)
         const files = params.files
-        console.log('upload-asset', data, files)
+        logger.info('upload-asset' + { data, files })
```
@speigg ::facepalms:: good catch, thanks. will fix

dmitrizagidulin replied:
@speigg fixed.

speigg commented:
@dmitrizagidulin I think you have to supply printf-style placeholders in pino to serialize additional parameters. See https://github.com/pinojs/pino/blob/master/docs/api.md#interpolationvalues-any

dmitrizagidulin commented:
@speigg - ok, changed my mind on adding the custom hook; refactored all the logger usage to fit the function signature.
(Apologies for the oversight, I was in a hurry)

speigg commented:
@dmitrizagidulin looks good, thank! Just waiting for CI/CD to finish and then I'll merge to dev!

speigg commented:
@dmitrizagidulin looks like we have a few typescript errors:

Error: @xrengine/analytics: ../server-core/src/hooks/party-user-permission-authenticate.ts(25,16): error TS2339: Property 'err' does not exist on type 'Logger<{ transport: { target: string; options: { colorize: boolean; translateTime: boolean; }; }; }>'.
Error: @xrengine/analytics: ../server-core/src/updateAppConfig.ts(348,75): error TS2552: Cannot find name 'message'. Did you mean 'onmessage'?


You can run `npm run check-errors` to reproduce locally. Can you fix these before we merge?
